pages/xogame.js

```diff
@@ -0,0 +1,192 @@
+            row.forEach((cell, columnIndex) => {
+                if (cell === "") {
+                    emptyIndexes.push({ rowIndex, columnIndex });
+                }
+            });
+        });
+        const randomIndex = Math.floor(Math.random() * emptyIndexes.length);
+        return emptyIndexes[randomIndex];
+    }
+
+    function checkWinner() {
+        for (let columnIndex = 0; columnIndex < board.length; columnIndex++) {
+            const row = board[columnIndex];
+            if (row.every((cell) => cell === cpuSymbol)) {
+                setWinner(cpuName);
+                return;
+            } else if (row.every((cell) => cell === playerSymbol)) {
+                setWinner(playerName);
+                return;
+            }
+        }
+
+        for (let i = 0; i < 3; i++) {
+            const column = board.map((row) => row[i]);
+            if (column.every((cell) => cell === cpuSymbol)) {
+                setWinner(cpuName);
+                return;
+            } else if (column.every((cell) => cell === playerSymbol)) {
+                setWinner(playerName);
+                return;
+            }
+        }
+
+        const diagonal1 = [board[0][0], board[1][1], board[2][2]];
+        const diagonal2 = [board[0][2], board[1][1], board[2][0]];
+        if (diagonal1.every((cell) => cell === cpuSymbol)) {
+            setWinner(cpuName);
+            return;
+        } else if (diagonal1.every((cell) => cell === playerSymbol)) {
+            setWinner(playerName);
+            return;
+        } else if (diagonal2.every((cell) => cell === cpuSymbol)) {
+            setWinner(cpuName);
+            return;
+        } else if (diagonal2.every((cell) => cell === playerSymbol)) {
+            setWinner(playerName);
+            return;
+        } else if (board.flat().every((cell) => cell !== "")) {
+            setWinner("draw");
+            return;
+        } else {
+            setWinner(null);
+            return;
+        }
+    }
+
+    function displayWinner() {
+        if (winner === "draw") {
+            return "Its a draw!";
+        } else if (winner) {
+            return `${winner} win!`;
+        }
+    }
+
+    function displayTurn() {
+        if (isCPUNext) {
+            return "Computers turn";
+        } else {
+            return "Your turn";
+        }
+    }
+
+    function resetPlayBoard() {
+        setBoard([
+            ["", "", ""],
+            ["", "", ""],
+            ["", "", ""],
+        ]);
+        setWinner(null);
+        setIsCPUNext(false);
+    }
+
+    return (
+        <>
+            <Main>
+                <h1>X and O Game</h1>
+                <h2>{!winner && displayTurn() || winner && displayWinner()}</h2>
+                <Matchfield>
+                    <Col />
+                    <Cell onClick={() => playFieldLocation(0, 0)} >
+                        {board[0][0]}
+                    </Cell>
+                    <Cell onClick={() => playFieldLocation(0, 1)} >
```

Comment:
If you map over the `board` array you could shorten this code dramatically.

Reply:
thanks but the game is already fully approved, the review went to the home page